Dialogue

DavidOk14 opened the pull request:
Haven't merged to main in a minute, gonna try and grind out a lot of what we can have left quest based and leave room for visual stuff that we can change prior to the final presentation.

Just letting y'all know where I'm at now as well as keeping main up to date, I'm aware I'll have to change the floor scenes to have all the same game objects while keeping the floor changing in tact.

DavidOk14 committed:
Definitely still needs some work to properly select the correct ink file (dialogue file) per character, will better organize it going onwards but is a small step forward for the deliverable as there is a working dialogue panel. Swapped player square for still player image, didn't implement animations fully but not of importance rn. Dialogue also needs to have a smaller size font, is currently using PressStart2P font at about 36pt. Also needs to be tweaked to either scroll or slowly reveal characters. TL;DR Working needs some adjustments

DavidOk14 committed:
Update FloorTwo.unity

DavidOk14 committed:
Dialogue added for NPCs [mostly, NPCs in this branch are currently tagged to one NPC still a bit iffy], player animation for pre-fab currently on idle and walking not for running, current bug with animation/movement, going to start placing NPCs on the first to get quest one out of the way for progression sakes onto my floor.

DavidOk14 committed:
Added some misc tiles for a few entrances of Discovery Park and for the cafeteria to help differentiate for when I add the cafeteria based NPCs, currently the goal is set for the first quest to be finding the keys which I'll work on later today to ensure the quest system is at least working prior to Neal. Also added music, and handling for walking/running/idle animations (a little buggy especially with the wall collision bug). The starter dialogue window appears and character can no longer move when dialogue text box appears, also yes the run speed is about a 5th faster than the regular walk speed right now. TL;DR a lot

DavidOk14 commented:
Forgot to mention but I also got rid of the directional arrow from the player prefab and the project all together since the character faces accordingly.

Kenneth-W-Chen commented on DP Mystery Map/Assets/Prefabs/DialoguePanel.prefab:
When you fix the merge conflicts, attach the gameobject to the "Gameplay Canvas" prefab and remove it from all of the scenes.

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scenes/FloorOne.unity:
When you fix the merge conflict, remove the camera, canvas, and player.

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scenes/FloorTwo.unity:
When you fix the merge conflict, remove the camera, canvas, and player.

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/DialogueManager.cs:
Can this not be attached directly to the DialoguePanel prefab? If you do that, you can link the script's serialized references to the prefab's children, and it should work without any effort besides instantiating the object

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/InteractivitySystem.cs:
When you fix the merge conflicts, replace the interactKey references with `Player.InteractKey`.

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/PlayerController.cs:
```diff
@@ -73,13 +73,15 @@ public void toggleMovement()
     private float _yMovement;
     private float _tanSpeed;
-    
+    private bool _isPlayerMoving;
```
Just use the `WalkingGrid` bool instead and rename it to something else. You also might need to add it to `FreeMoveFixedUpdate()`

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/PlayerController.cs:
```diff
@@ -73,13 +73,15 @@ public void toggleMovement()
     private bool _walkOn = false;
     private bool _walkOnce = false;
     private float _walkStartTime;
+    private bool _isPlayerColliding = false;
```
Please don't work on tasks that you weren't assigned. I'm in charge of fixing that bug, and I already finished work on it, so remove this

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/PlayerController.cs:
```diff
@@ -137,12 +146,17 @@ void Start()
             { Direction.Right, Player.MoveRightKey }
         };
+
+        Player.FacingDirection = Direction.Down;
```
Remove this. This forces the player to face downwards even though save files save the player's facing direction. If this is for the animator, then have the animator itself change in `Start()`

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/PlayerController.cs:
```diff
@@ -137,12 +146,17 @@ void Start()
     {
-        _movePlayerUpdate();
+        //Only allow character movement when dialogue isn't active
+        if(!DialogueManager.dialogIsPlaying)
```
When you fix the merge conflicts, remove this check here and set the `WalkBlocked` property with `WalkBlockedFlags.Interacting` in DialogueManager.

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/PlayerController.cs:
```diff
@@ -208,13 +222,83 @@ private void GridMoveUpdate()
             }
 
+            //Update the idleFacing position per direction
+            switch (Player.FacingDirection)
```
Put this in a new `DirectionChangeEventHandler` and subscribe it to the `Player.directionChangeEvent` event. Make sure to unsubscribe the event in the `PlayerController.OnDestroy()` method too.

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/PlayerController.cs:
```diff
@@ -208,13 +222,83 @@ private void GridMoveUpdate()
+            else
+            {
+            //Enable running animation if shift is held
+            if (Input.GetKey(KeyCode.LeftShift) && ((Input.GetKey(Player.MoveUpKey)) ||
```
David this task wasn't assigned to you in the production plan. Also, this introduces a bug where the player can change their walk speed in the middle of a step.

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/PlayerController.cs:
```diff
@@ -208,13 +222,83 @@ private void GridMoveUpdate()
+            switch (_walkDirection)
+                {
+                    case Direction.Left:
+                        playerMovementController.x = Input.GetAxisRaw("Horizontal");
```
Don't use `GetAxisRaw()`. If the player presses the left arrow or A, then `GetAxisRaw("Horizontal")` returns -1. Right arrow or D returns 1. And so on. `GetAxisRaw` won't work properly if the player changes the movement keys. Just set the Vector2 to the predefined static Vector2 (e..g, `Vector2.left`).
Put this in a new `DirectionChangeEventHandler` and subscribe it to the `Player.directionChangeEvent` event. Make sure to unsubscribe the event in the `PlayerController.OnDestroy()` method too. **You may consider putting all of the animator event subscribers in one function.**

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/PlayerController.cs:
```diff
@@ -365,4 +448,19 @@ private void UnsubscribeEvents()
     }
-} 
+
+    private void OnCollisionEnter2D(Collision2D collision)
```
Again, remove these since the issue is already fixed in a commit on main

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/questSystem.cs:
```diff
@@ -0,0 +1,24 @@
+    // Start is called before the first frame update
+    void Start()
+    {
+        if(Player.collectedItems == Item.None)
```
This breaks if the player enters the game on floor 2 without any items (i.e., the player saves and loads the game on floor 2 without collecting any items).

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/DialogueManager.cs:
```diff
@@ -0,0 +1,110 @@
+    private string fullText;
+    private float charRevealDelay = 0.05f;
+
+    public static DialogueManager instance { get; private set; }
```
Add `OnDestroy()` and make it set `instance` to if `instance` is `this`. Otherwise, the DialogueManager can't be created after the first one is destroyed

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/DialogueManager.cs:
```diff
@@ -0,0 +1,110 @@
+            return;
+
+
+        if (Input.GetKeyUp(InteractivitySystem.interactKey) && (dialogText.text == fullText))
```
Could you change it so that a different key(s) advance the text? Make it a separate static variable in `Player`. Use a list or have 3-4 variables (default keys like space and enter, the left mouse button, and then a player-set key).

Kenneth-W-Chen commented on DP Mystery Map/Assets/Scripts/DialogueManager.cs:
```diff
@@ -0,0 +1,110 @@
+        for (int i = 0; i < fullText.Length; i++)
+        {
+            dialogText.text = fullText.Substring(0, i + 1);
+            yield return new WaitForSeconds(charRevealDelay);
```
This is more of a QoL than a requirement for the review, but would you be able to make the text advance immediately if the player presses one of the keys that continues the text? So if the dialogue is still displaying and the player presses the interact key, make the text display fully immediately

Kenneth-W-Chen commented on DP Mystery Map/Assets/TextMesh Pro/Documentation.meta:
Delete this

Kenneth-W-Chen commented on DP Mystery Map/Assets/TextMesh Pro/Documentation/TextMesh Pro User Guide 2016.pdf:
Delete this

Kenneth-W-Chen commented on DP Mystery Map/Assets/TextMesh Pro/Documentation/TextMesh Pro User Guide 2016.pdf.meta:
Delete this

Kenneth-W-Chen commented on DP Mystery Map/Assets/Ink/CHANGELOG.md:
Delete this

Kenneth-W-Chen commented on DP Mystery Map/Assets/Ink/CHANGELOG.md.meta:
Delete this

Kenneth-W-Chen commented on DP Mystery Map/Assets/Ink/Demos.meta:
Delete the entire Demos folder. We don't need that

Kenneth-W-Chen commented on DP Mystery Map/Assets/Ink/Documentation.meta:
I'd prefer if you delete the entire Documentation folder too. It's just more project bloat as is

Kenneth-W-Chen commented on DP Mystery Map/Assets/Ink/README.md:
You don't need to delete this.

Kenneth-W-Chen commented on DP Mystery Map/Assets/Fonts/PressStart2P SDF.asset:
Would you mind tossing this asset into the `TextMesh Pro/Fonts` folder?

Kenneth-W-Chen reviewed (changes requested):
I've left comments on multiple files for stuff to change. Also, the walk animation doesn't play if the player taps a movement key to walk. Don't fix it if it's going to take more than an hour, unless you don't have another task to work on.

Also if you're moving/renaming/deleting files, do them in Unity's file explorer window. That way the meta files update.

DavidOk14 committed:
Adjusted changes as well as I could, made it so spacebar advances the text in the dialogue system, it's a little buggy and works sometimes. Don't have the gameplay canvas pre-fab so I'll attach the gameobject once I merge and delete the camera from both scenes.

DavidOk14 committed:
Changes

DavidOk14 commented:
Took some time to get the pull request done because of work, besides being able to pick up a key right now I'm not sure what else I'll be able to get down before the presentation

Kenneth-W-Chen committed:
Fix Player class merge conflicts

Kenneth-W-Chen committed:
Fix InteractivitySystem merge conflict with main

Kenneth-W-Chen committed:
Add flag for game events/quests

Kenneth-W-Chen committed:
Add event flags property and field

Kenneth-W-Chen committed:
Add data saving for game event flags

Kenneth-W-Chen committed:
Added a function for setting event flags to make it easier to use

Kenneth-W-Chen committed:
Implement event flag to show opening scene only once

Kenneth-W-Chen committed:
Fixed a variable that was renamed

Issue came from the merge conflict resolution in Player.cs

DavidOk14 committed:
Added the mini-map camera since the pre-fab only attached a black box, Scene two doesn't load in with the camera displaying properly. Dialogue isn't advancing, added a boolean for if the player started a new game so the initial dialogue plays, gets set to false after so the initial dialogue doesn't continuously play similarly to before.

Kenneth-W-Chen committed:
replace with actual file

Kenneth-W-Chen committed:
Fix merge conflicts

Kenneth-W-Chen committed:
Fix file locations and references

Kenneth-W-Chen committed:
Fixed some bugs.

Fixed player being able to move during opening scene dialogue. Fixed dialogue not showing. Adjusted Dialogue Panel's size

Kenneth-W-Chen committed:
Fix collider sizes

Kenneth-W-Chen committed:
Fixed sprite sizing

Kenneth-W-Chen committed:
Fix tags

Kenneth-W-Chen committed:
Fix tag

Kenneth-W-Chen committed:
Add dialogue manager

Kenneth-W-Chen committed:
Fix reference to old variable

Kenneth-W-Chen committed:
Fixed bug that prevented the text fast forward

Also changed some keybindings

Kenneth-W-Chen committed:
Fix bug that prevent stairs from working

Kenneth-W-Chen committed:
Update FloorTwo to most recent version

Kenneth-W-Chen committed:
Fix wall Tags

Kenneth-W-Chen committed:
Fixed ordering of objects

Kenneth-W-Chen committed:
Fixed stairs walking issue

Kenneth-W-Chen committed:
Fixed stairs walking issue

Kenneth-W-Chen committed:
Fixed a bug when loading game

Bug caused the dialogue panel to show regardless of what was happening

DavidOk14 committed:
Dialogue across all current NPCs

DavidOk14 committed:
Dialogue across all current NPCs

DavidOk14 committed:
Final touches

Kenneth-W-Chen commented:
What did you add that pushed FloorOne over 100mb? I'm pretty sure it was only 43mb when I committed the last change to this branch

If it's under 100mb, then don't push it with lfs. If you want to use lfs, put a Google Drive/OneDrive link in the readme with the file
